source/SH3/camera/camera.cpp

```diff
@@ -0,0 +1,131 @@
+/** @file
+ *
+ *  Implementation of camera.hpp
+ *
+ *  @copyright 2017 Palm Studios
+ *
+ *  @date 1-9-2017
+ *
+ *  @author Jesse Buhagiar
+ */
+#include <cmath>
+#include "SH3/camera/camera.hpp"
+#include "glm/gtc/matrix_transform.hpp"
+#include <algorithm>
+#include <boost/algorithm/clamp.hpp>
+
+static constexpr glm::vec3 worldUp = glm::vec3(0, 1, 0); /**< Which axis is considered 'world up'. */
+
+sh3::camera::Camera::Camera(const glm::vec3& pos, angle<float> fov, float aspect, float zNear, float zFar)
+: camPos(pos), camFov(fov), aRatio(aspect), camNear(zNear), camFar(zFar),
+  camFront(glm::vec3(0.0f, 0.0f, -1.0f)), camUp(glm::vec3(0.0f, 1.0f, 0.0f)),
+  camPitch(angle<float>::FromDegrees(-90.0f)),
+  camYaw(angle<float>::FromDegrees(0.0f)),
+  camMode(MODE::FIRST_PERSON)
+{
+    RecalculatePerspective();
+}
+
+void sh3::camera::Camera::RecalculatePerspective()
+{
+    glm::vec3 front;
+    front.x = std::cos(camPitch.AsRadians()) * std::cos(camYaw.AsRadians());
+    front.y = std::sin(camPitch.AsRadians());
+    front.z = std::cos(camPitch.AsRadians()) * std::sin(camYaw.AsRadians());
+    camFront = glm::normalize(front);
+
+    camRight = glm::normalize(glm::cross(camFront, worldUp));
+    camUp = glm::normalize(glm::cross(camRight, camFront));
+
+    vMatrix = glm::lookAt(camPos, camPos + camFront, camUp);
```

Comment:
vMatrix is only used here. Can it be removed? Or if `pMatrix` stores the product as suggested below, `vMatrix` doesn't need to remain a member of `Camera`.